nevergrad/functions/gym/gym_multi.py

```diff
@@ -0,0 +1,180 @@
+#         pass
+
+gym_env_names = [
+    "gym_anm:ANM6Easy-v0",
+    "Copy-v0",
+    "RepeatCopy-v0",
+    "ReversedAddition-v0",
+    "ReversedAddition3-v0",
+    "DuplicatedInput-v0",
+    "Reverse-v0",
+    "CartPole-v0",
+    "CartPole-v1",
+    "MountainCar-v0",
+    "Acrobot-v1",
+    "Blackjack-v0",
+    "FrozenLake-v0",
+    "FrozenLake8x8-v0",
+    "CliffWalking-v0",
+    "NChain-v0",
+    "Roulette-v0",
+    "Taxi-v3",
+    "CubeCrash-v0",
+    "CubeCrashSparse-v0",
+    "CubeCrashScreenBecomesBlack-v0",
+    "MemorizeDigits-v0",
+]
+
+
+class GymMulti(ExperimentFunction):
+    def __init__(
+        self, name: str = "gym_anm:ANM6Easy-v0", control: str = "conformant", neural_factor: int = 2
+    ) -> None:
+        env = gym.make(name)
+        self.name = name + "__" + control + "__" + str(neural_factor)
+        self.num_time_steps = 100
+        self.neural_factor = neural_factor
+        o = env.reset()
+        if "int" in str(type(env.action_space.sample())):  # Discrete action space
+            output_dim = env.action_space.n
+            output_shape = (output_dim,)
+            discrete = True
+        else:  # Continuous action space
+            output_shape = tuple(np.asarray(env.action_space.sample()).shape)  # type: ignore
+            discrete = False
+            output_dim = np.prod(output_shape)
+        if "int" in str(type(o)):
+            input_dim = env.observation_space.n
+            self.discrete_input = True
+        else:
+            input_dim = np.prod(np.asarray(o).shape)
+            self.discrete_input = False
+        self.output_shape = output_shape
+        self.input_dim = input_dim
+        self.output_dim = output_dim
+        self.num_neurons = neural_factor * input_dim
+        assert control in [
+            "conformant",
+            "linear",
+            "neural",
+            "noisy_neural",
+            "noisy_scrambled_neural",
+            "scrambled_neural",
+        ], f"{control} not known as a form of control"
+        self.control = control
+        neural_size = (output_dim * self.num_neurons + self.num_neurons * (input_dim + 1),)
+        shape = {
+            "conformant": (self.num_time_steps,) + output_shape,
+            "linear": (input_dim + 1, output_dim),
+            "neural": neural_size,
+            "noisy_neural": neural_size,
+            "noisy_scrambled_neural": neural_size,
+            "scrambled_neural": neural_size,
+        }[control]
+        shape = tuple(int(s) for s in shape)
+        self.policy_shape = shape
+        parametrization = parameter.Array(shape=shape)
+        super().__init__(self.gym_multi_function, parametrization=parametrization)
+        self.env = env
+        self.discrete = discrete
+
+    def env_names(self):
+        return gym_env_names
+
+    def discretize(self, a):
+        probabilities = np.exp(a - max(a))
+        probabilities = probabilities / sum(probabilities)
+        return np.random.multinomial(1, probabilities)[0]
```

Comment:
why not use the parametrization directly, doing this is duplicating the code + losing the information which should have been useful for the optimizer

Reply:
It would not work! More precisely that would work only in the case of conformant planning, so I think it's better to not do it.

Reply:
you'll need to explain this to me, because that's not clear to me. In my view the parametrization is supposed to describe the problem

Reply:
oh, because, the think that is discretize is not directly the output of nevergrad? Eg nevergrad looks for neural nets and the discretization is on the output?